StreamChatClient_v3/APIClient/Endpoints/ChannelEndpoints.swift

```diff
@@ -21,4 +21,28 @@ extension Endpoint {
               requiresConnectionId: query.options.contains(oneOf: [.presence, .state, .watch]),
               body: query)
     }
+
+    static func delete<ExtraData: ExtraDataTypes>(channel: ChannelModel<ExtraData>) -> Endpoint<EmptyResponse> {
+        .init(path: "channels/\(channel.cid.type)/\(channel.cid.id)",
+              method: .delete,
+              queryItems: nil,
+              requiresConnectionId: false,
+              body: nil)
+    }
+
+    static func hide<ExtraData: ExtraDataTypes>(channel: ChannelModel<ExtraData>, userId: UserId, clearHistory: Bool) -> Endpoint<EmptyResponse> {
+        .init(path: "channels/\(channel.cid.type)/\(channel.cid.id)/hide",
+              method: .post,
+              queryItems: nil,
+              requiresConnectionId: false,
+              body: HideChannelRequest(userId: userId, clearHistory: clearHistory))
+    }
+
+    static func show<ExtraData: ExtraDataTypes>(channel: ChannelModel<ExtraData>, userId: UserId) -> Endpoint<EmptyResponse> {
+        .init(path: "channels/\(channel.cid.type)/\(channel.cid.id)/show",
+              method: .post,
+              queryItems: nil,
+              requiresConnectionId: false,
+              body: ["userId": userId])
+    }
```

Comment:
- Ideally, the function won't take the whole `ChannelModel` but only `ChannelId`.
- Please add tests. This will create some conflicts with this https://github.com/GetStream/stream-chat-swift/pull/378#discussion_r465800314 so if you guys could coordinate it would be great.

Reply:
Yep, I've done it that way in first place but later changed this for prettier function calls. `hide(channel:)` seems nicer than `hideChannel(channelId:)`. What do you think?

Reply:
I see what you mean, but here's it's actually not about API design. `ChannelModel` is just an immutable snapshot of data in the DB. It's different from v2. You can't just create it, you always need to fetch it from the db. It's unnecessary work when all we really need is `cid`.

Also, please always use `cid` when referring to variables of type `ChannelId`. I know it might sound strange but it's because of the parity with other SDKs and our backend and documentation.

Reply:
Ok, got it!